networkx/algorithms/efficiency.py

```diff
@@ -0,0 +1,140 @@
+# efficiency.py - functions for computing node, edge, and graph efficiency
+#
+# Copyright 2011, 2012, 2013, 2014, 2015 NetworkX developers
+#
+# This file is part of NetworkX.
+#
+# NetworkX is distributed under a BSD license; see LICENSE.txt for more
+# information.
+"""Provides functions for computing the efficiency of nodes and graphs."""
+from __future__ import division
+
+from itertools import combinations
+
+import networkx as nx
+from ..utils import not_implemented_for
+
+__all__ = ['efficiency', 'local_efficiency', 'global_efficiency']
+
+
+@not_implemented_for('directed')
+def efficiency(G, u, v):
+    """Returns the efficiency of a pair of nodes in a graph.
+
+    The *efficiency* of a pair of nodes is the multiplicative inverse of the
+    shortest path distance between the nodes [1]_.
+
+    Parameters
+    ----------
+    G : :class:`networkx.Graph`
+        An undirected graph for which to compute the average local efficiency.
+    u, v : node
+        Nodes in the graph ``G``.
+
+    Returns
+    -------
+    float
+        Multiplicative inverse of the shortest path distance between the nodes.
+
+    Notes
+    -----
+    Edge weights are ignored when computing the shortest path distances.
+
+    See also
+    --------
+    local_efficiency
+    global_efficiency
+
+    References
+    ----------
+    .. [1] Latora, Vito, and Massimo Marchiori.
+           "Efficient behavior of small-world networks."
+           *Physical Review Letters* 87.19 (2001): 198701.
+           http://dx.doi.org/10.1103/PhysRevLett.87.198701
+
+    """
+    return 1 / nx.shortest_path_length(G, u, v)
```

Comment:
You probably want `1. / nx.shortest_path_length(G, u, v)` to avoid truncation, right? If the shortest path length function returns an integer, the division will also return an int.